Sources/APPX.cpp

```diff
@@ -102,6 +104,15 @@ namespace appx {
             for (const auto &fileNamePair : fileNames) {
                 const std::string &archiveName = fileNamePair.first;
                 const std::string &fileName = fileNamePair.second;
+
+                const std::string suffix = "AppxBundleManifest.xml";
+                if (bundle &&
+                        suffix.size() < archiveName.size() &&
+                        std::equal(suffix.rbegin(), suffix.rend(), archiveName.rbegin())) {
```

Comment:
That is fair, we have enough info in the .appx packages to create the AppxBundleManifest.xml directly from the appx executable, but it would be a lot of work to do it from the appx executable (I am doing this in the python script atm)

As for the comment on "We are doing the same thing with AppxManifest.xml for .APPX archives." - the python script (windows_package_app.py) provides a pair of files for AppxManifest.xml and also StoreManifest.xml and they are provided in the same way all other files are provided. Here I am assuming that AppxManifest.xml is needed for the appx files because at least in all our cases we provide it. 

If we are going to change the way we treat these manifests, we would need to change it when we create both appx and appxbundle. Maybe we can take this offline.